src/components/SharedComponents/Buttons/PressableWithDebounce.js

```diff
@@ -0,0 +1,77 @@
+import {
+  Pressable
+} from "components/styledComponents";
+import React, { useEffect, useRef, useState } from "react";
+import { GestureResponderEvent, ViewStyle } from "react-native";
+
+interface Props {
+    accessibilityHint?: string;
+    accessibilityLabel?: string;
+    accessibilityRole?: string;
+    style?: ViewStyle;
+    className?: string;
+    onPress: () => void;
+    testID?: string;
+    children?: unknown;
+    disabled: boolean;
+    debounceTime?: number;
+    preventMultipleTaps?: boolean;
+}
+
+const PressableWithDebounce = ( {
+  accessibilityHint,
+  accessibilityLabel,
+  accessibilityRole,
+  style,
+  className,
+  testID,
+  onPress,
+  disabled,
+  children,
+  debounceTime = 400,
+  preventMultipleTaps = true
+}: Props ) => {
+  const [isProcessing, setIsProcessing] = useState( false );
+  const onPressRef = useRef( onPress );
+
+  useEffect( ( ) => {
```

Comment:
One small change requested here -- can you try this without the `useEffect`? so just directly set `onPressRef.current = onPress` underneath `onPressRef`. That worked for our related PR here, and should alleviate Johannes' performance concerns: https://github.com/inaturalist/iNaturalistReactNative/pull/2844/files